tee-worker/omni-executor/aa-contracts/aa-demo-app/src/components/AuthorizeTEEWorker.tsx

```diff
@@ -0,0 +1,280 @@
+
+	const handleAuthorize = async () => {
+		if (
+			!omniAccountAddress ||
+			!omniAccountHash ||
+			!publicClient ||
+			!walletClient ||
+			!evmAddress ||
+			!isDeployed
+		)
+			return;
+
+		setIsAuthorizing(true);
+		setError(null);
+
+		try {
+			// Step 1: Authenticate with TEE worker and get its address
+			const { workerAddress: teeWorkerAddress } = await authorizeTEEWorker(
+				walletClient,
+				evmAddress,
+				TEE_WORKER_CONFIG.clientId,
+				omniAccountHash
+			);
+
+			setWorkerAddress(teeWorkerAddress);
+
+			// Step 2: Add the TEE worker as an authorized signer
+			setIsAddingSigner(true);
+
+			// First encode the addRootSigner call
+			const addSignerData = encodeFunctionData({
+				abi: CONTRACTS.OmniAccountImplementation.abi,
+				functionName: "addRootSigner",
+				args: [teeWorkerAddress as `0x${string}`],
+			});
+
+			// Then wrap it in an execute call (the account calls itself)
+			const callData = encodeFunctionData({
+				abi: [
+					{
+						name: "execute",
+						type: "function",
+						inputs: [
+							{ name: "target", type: "address" },
+							{ name: "value", type: "uint256" },
+							{ name: "data", type: "bytes" },
+						],
+						outputs: [],
+						stateMutability: "nonpayable",
+					},
+				],
+				functionName: "execute",
+				args: [omniAccountAddress as `0x${string}`, BigInt(0), addSignerData],
+			});
+
+			// Get current nonce for the account
+			const nonce = (await publicClient.readContract({
+				address: CONTRACTS.EntryPoint.address,
+				abi: CONTRACTS.EntryPoint.abi,
+				functionName: "getNonce",
+				args: [omniAccountAddress as `0x${string}`, BigInt(0)],
+			})) as bigint;
+
+			// Create UserOperation
+			const userOp = createUserOperation({
+				sender: omniAccountAddress as `0x${string}`,
+				nonce,
+				callData,
+				initCode: "0x", // Account already deployed
+			});
+
+			// Sign UserOperation with RootKey signer type
+			const signature = await signUserOperation(
+				walletClient,
+				evmAddress,
+				userOp as UserOperation,
+				CONTRACTS.EntryPoint.address,
+				BigInt(await publicClient.getChainId()),
+				UserOpSigner.RootKey,
+			);
+
+			userOp.signature = signature;
+
+			// Convert to PackedUserOperation for EntryPoint v0.7
+			const packedUserOp = packUserOperation(userOp as UserOperation);
+
+			// Send UserOperation via EntryPoint
+			const hash = await walletClient.writeContract({
+				address: CONTRACTS.EntryPoint.address,
+				abi: CONTRACTS.EntryPoint.abi,
+				functionName: "handleOps",
+				args: [[packedUserOp], evmAddress] as const,
+				chain: walletClient.chain,
+				account: walletClient.account!,
+			});
+
+			// Wait for transaction
+			const receipt = await publicClient.waitForTransactionReceipt({ hash });
+			console.log("Transaction receipt:", receipt);
+
+			// Add a delay to ensure events are indexed
+			console.log("Waiting for events to be indexed...");
+			await new Promise((resolve) => setTimeout(resolve, 2000)); // 2 second delay
```

Comment:
Just a suggestion, it doesn't necessarily need to be changed, after all, this is a demo, as long as the effect can be achieved.

`waitForTransactionReceipt` has a comfirmed status, no need to wait for a delay.  , and `writeContract` can be placed in the same hooks as `waitForTransactionReceipt`, making it convenient for external calls and monitoring transaction status. I used it [this way ](https://github.com/0xverin/Nextjs-Wagmi-QuickStart/blob/main/hooks/contracts/useTestTokenContract.ts#L97)before.